src/lunamary/services/SchoolService.java

```diff
@@ -1,4 +1,176 @@
 package lunamary.services;
 
+import datastructures.arraylist.MyArrayList;
+import datastructures.hashmap.MyHashMap;
+import lunamary.ReadWriteData.AbstractFactory;
+import lunamary.ReadWriteData.ReadWriteFile;
+import lunamary.modelPerson.*;
+import lunamary.modelSchool.*;
+;import java.io.FileNotFoundException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
 public class SchoolService {
+
+    private static School school;
+    private AbstractFactory factory;
+
+
+    public SchoolService() {
+
+
+    }
+
+    public School createSchool(String name, String address) {
```

Comment:
This is not a proper implementation of Singleton design pattern, please check some documentation.
Implementing this correctly will simplify a lot of the logic